en/docs/install-and-setup/setup/api-controller/managing-microgateways/managing-microgateways-with-ctl.md

```diff
@@ -0,0 +1,296 @@
+        ```  
+
+    !!! info
+        **Flags:**
+        
+        - Mandatory :    
+            `-e`, `--environment` : Microgateway adapter environment to add the API   
+            `-f`, `--file`        : Filepath of the apictl project to be deployed
+
+        - Optional :     
+            `-o`, `--override` : Whether to deploy an API irrespective of its existance. Overrides when exists.    
+            `--skip-cleanup` : Whether to keep all temporary files created during the deploy process
+
+    !!! example
+        ```bash
+        apictl mg deploy api -e dev -f petstore -k
+        ```
+        ```bash
+        apictl mg deploy api --environment dev --file petstore -k
+        ```
+        ```bash
+        apictl mg deploy api -e dev -f petstore --override -k
+        ```
+        ```bash
+        apictl mg deploy api -e dev -f petstore -o --skip-cleanup -k
+        ```
+
+-   **Response**
+
+    ``` bash tab="Deployed"
+    Successfully deployed API to microgateway.
+    ```
+
+    ``` bash tab="API Already Exists"
+    Unable to deploy API. API already exists. Status: Conflict 409
+    ```
+
+    ```bash tab="Deployed or Updated"
+    Successfully deployed/updated the API in microgateway.
+    ```
+
+## List Deployed APIs
+
+-   **Command**
+    ```bash
+    apictl mg get apis -e <environment-name> -k
+    ```
+    !!! tip
+        By default, the number of APIs listed will be limited to 25. To increase or decrease the limit set the flag `--limit` or its shorthand flag `-l`. For an example,
+        ```bash
+        apictl mg get apis -e dev --limit 100 -k
+        ```
+
+    !!! tip
+        Currently, these APIs can be filtered by API type. The supported types are http and ws (web socket). Use the flag `--query` or its shorthand flag `-q` to filter APIs.
+        ```bash
+        apictl mg get apis -e dev --query type:http -k
+        ```
+
+    !!! info
+        **Flags:**
+        
+        - Mandatory :    
+            `-e`, `--environment` : Microgateway adapter environment to list the APIs from  
+
+        - Optional :       
+            `-l`, `--limit` : Maximum number of APIs to return     
+            `-q`, `--query` : Query to filter the APIs
+
+    !!! example
+        ```bash
+        apictl mg get apis -e dev -k
+        ```
+        ```bash
+        apictl mg get apis -q type:http --environment dev --limit 100 -k
+        ```
+        ```bash
+        apictl mg get apis -q type:ws -e dev -l 10 -k
+        ```
+
+-   **Response**
+
+    ```
+    Limit flag not set. Set to default: 25
+    APIs total: 2 received: 2
+    NAME                VERSION             TYPE                CONTEXT         GATEWAY_ENVS
+    SwaggerPetstore     1.0.5               HTTP                /v2             [Production and Sandbox]
+    SwaggerPetstore1    1.0.5               WS                  /v2             [Production and Sandbox]
+    ```
+
+
+## Undeploy an API
+
+This lets you remove an API from the microgateway. 
```

Comment:
Shall we add something like this as the introduction of this command?
```suggestion
This command uses to remove an API from the microgateway. 
```

Reply:
@Chamindu36 shall we change this like below.
"This command can be used to remove an API from the microgateway."
WDYT?

Reply:
+1

Reply:
Updating the line to "This command can be used to remove an API from the microgateway."